src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -1019,6 +1018,20 @@ cpdef object compute_network_structured(
                 #tuple of MC_Reach and reach_type
                 MC_Reach(segment_objects, array('l',upstream_ids))
                 )
+            
+    cdef np.ndarray fill_index_mask = np.ones_like(data_idx, dtype=bool)
+    cdef Py_ssize_t fill_index
+    cdef long upstream_tw_id
+    cdef dict tmp
+    cdef int idx
+    cdef float val
+    
+    for upstream_tw_id in upstream_results:
+    tmp = upstream_results[upstream_tw_id]
+    fill_index = tmp["position_index"]
+    fill_index_mask[fill_index] = False
+    for idx, val in enumerate(tmp["results"]):
+        flowveldepth_nd[fill_index, (idx/3)+1] = val
```

Comment:
idx/3 could result in a float and lead to an error here.

Reply:
@awlostowski-noaa The solution might be to use integer division with `//` 
(see [here](https://betterprogramming.pub/lesser-known-arithmetic-operators-in-python-a34670087b3a))

Reply:
The second dimension is the timesteps, so I would imagine you would want `(idx + 1) * 3` assuming that `len(tmp['results']) == number of timesteps`